### PR TITLE
Hookup FixedByteChunkSingleValue indexing.

### PR DESCRIPTION
**Pull request by mayankshriv**
1. Hooked up FixedButeChunkSingleValue indices into data
   fetching during query processing. Added reading of
   int, long and float data types from this index.

2. Fixed bug in FixedByteChunkSingleValueWriter.

3. Added unit test for FixedByteChunkSingleValueReader/Writer.
   Also, enhanced DataFetcherTest to support no-dictionary index.

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/common/DataFetcher.java [hunk 1]
```diff
@@ -177,8 +177,13 @@ public int getMaxNumberOfEntriesForColumn(String column) {
...
   public void fetchIntValues(String column, int[] inDocIds, int inStartPos, int length, int[] outValues, int outStartPos) {
     Dictionary dictionary = getDictionaryForColumn(column);
-    fetchSingleDictIds(column, inDocIds, inStartPos, length, _reusableDictIds, 0);
-    dictionary.readIntValues(_reusableDictIds, 0, length, outValues, outStartPos);
+    if (dictionary != null) {
+      fetchSingleDictIds(column, inDocIds, inStartPos, length, _reusableDictIds, 0);
+      dictionary.readIntValues(_reusableDictIds, 0, length, outValues, outStartPos);
+    } else {
+      BlockValSet blockValSet = _columnToBlockValSetMap.get(column);
+      blockValSet.getIntValues(inDocIds, 0, length, outValues, outStartPos);
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/com/linkedin/pinot/core/common/DataFetcher.java]: The second argument should be inStartPos?
Same for all other methods.

File: pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java [hunk 2]
```diff
@@ -68,37 +68,44 @@ public FixedByteChunkSingleValueWriter(File file, ChunkCompressor compressor, in
...
   public void setDouble(int row, double value) {
-    _chunkBuffer.putDouble(_chunkDataOffset, value);
+    _chunkBuffer.putDouble(value);
     _chunkDataOffset += DOUBLE_SIZE;
     flushChunkIfNeeded();
   }
 
+  @Override
+  protected void writeChunk() {
+    _chunkBuffer.flip();
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java]: Please move _chunkBuffer.flip() into base class writeChunk() method. (Also remove the _chunkBuffer.flip() inside VarByteChunkSingleValueWriter writeChunk())

File: pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java [hunk 3]
```diff
@@ -68,37 +68,44 @@ public FixedByteChunkSingleValueWriter(File file, ChunkCompressor compressor, in
...
+    super.writeChunk();
+    _chunkBuffer.clear();
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java]: No need to clear() as it's already cleared in base writeChunk(). Just revert this part and move _chunkBuffer.flip() into base class should be good.

File: pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java [hunk 4]
```diff
@@ -68,37 +68,44 @@ public FixedByteChunkSingleValueWriter(File file, ChunkCompressor compressor, in
 
   @Override
   public void setInt(int row, int value) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java]: General question: argument row is ignored, so the behavior of this writer is different with old writers which implements SingleColumnSingleValueWriter interface. Do you plan to handle that?

File: pinot-core/src/main/java/com/linkedin/pinot/core/operator/docvalsets/UnSortedSingleValueSet.java [hunk 5]
```diff
@@ -46,6 +46,105 @@ public DataType getValueType() {
...
+   * @param outValues Array where output is written
+   * @param outStartPos start index into output array
+   */
+  @Override
+  public void getFloatValues(int[] inDocIds, int inStartPos, int inDocIdsSize, float[] outValues, int outStartPos) {
+    int inEndPos = inStartPos + inDocIdsSize;
+    ReaderContext context = sVReader.createContext();
+
+    switch (columnMetadata.getDataType()) {
+      case INT:
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/com/linkedin/pinot/core/operator/docvalsets/UnSortedSingleValueSet.java]: IMO, should never get float values from INT or LONG?

Comment (mayankshriv) [inline pinot-core/src/main/java/com/linkedin/pinot/core/operator/docvalsets/UnSortedSingleValueSet.java]: I went with whatever is allowed in java without casting.

File: pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java (hunk 4 above)
Comment (mayankshriv) [inline pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java]: Yeah, chunk based writers with compression do support writes at random offsets. Updated javadoc with the same.

File: pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java (hunk 2 above)
Comment (mayankshriv) [inline pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java]: Done

File: pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java (hunk 3 above)
Comment (mayankshriv) [inline pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/v1/FixedByteChunkSingleValueWriter.java]: Done

File: pinot-core/src/main/java/com/linkedin/pinot/core/common/DataFetcher.java (hunk 1 above)
Comment (mayankshriv) [inline pinot-core/src/main/java/com/linkedin/pinot/core/common/DataFetcher.java]: Done

**Pull request merged**